packages/components/Menu/src/MenuItem/useMenuItem.ts

```diff
@@ -99,12 +114,23 @@ function getAccessibilityStateWorker(disabled: boolean, accessibilityState?: Acc
   return { disabled };
 }
 
-export const useHoverFocusEffect = (hovered: boolean, componentRef: React.MutableRefObject<any>) => {
+export const useHoverFocusEffect = (hovered: boolean, componentRef: React.MutableRefObject<any>, setIsMenuItemHovered: any) => {
   React.useLayoutEffect(() => {
     if (hovered) {
       componentRef?.current?.focus();
+      setIsMenuItemHovered(true);
     } else {
       componentRef?.current?.blur();
+      setIsMenuItemHovered(undefined);
+    }
+  }, [hovered, componentRef, setIsMenuItemHovered]);
+};
+
+export const useRemoveHoverEffect = (hovered: boolean, isMenuItemHovered: boolean, onHoverOut: any) => {
+  React.useLayoutEffect(() => {
+    if (hovered && isMenuItemHovered != undefined && !isMenuItemHovered) {
+      // remove Focus from this menuITem
+      onHoverOut();
```

Comment:
I had a similar comment in a previous PR (not one you made). 
Manually calling `onHoverOut` feels like tricking React Native that mouse hover is gone, but doesn't actually reflect the native state (you could call `onHoverOut` while the mouse is over the component for example, and now native / RN state are out of sync). Why do we need to manually call it?

Reply:
That's exactly what we are trying to do actually, calling onHoverOut while the mouse is over the component but keyboard focus is on a different component. Then, even if the mouse is still over the component, as long as you're keyboard focusing through other components, we don't want hover sate.

Reply:
On macOS, we do that by not having any special styling for `hover`. We only set tokens for `focus` and `onHoverIn/Out` calls `ref.current.focus` via the `useFocusHoverEffect` hook. We effectively don't use hover to inform any styling, focus is the "one" source of truth, and hover updates focus. https://github.com/microsoft/fluentui-react-native/blob/4430c6d93553e25870ab350fb7304adbfd2d456b/packages/components/Menu/src/MenuItem/MenuItemTokens.macos.ts#L23-L27

Is there different styling for hover state and focus state on menu items in win32? On macOS they look the same. I had assumed till now that on win32, you could have a menu item hovered and another menu item keyboard focused and they would have different looking selection states. It sounds like from this PR, we actually only want one state: selected. If so, would what we do on macOS work for win32?

Reply:
or (maybe too much for this PR) have both hover and focus update some state at the MenuList level of the currently "selected" menu item? That way the MenuList is what manages styling, and each menu item is just using its callbacks to update the entire menu's state?

Reply:
We actually do have separate styling defined for hovered state. Focused state and hovered state styles are identical. Saad, I think you just made me realize this could have been a one liner fix LOL